src/neurostatslib/solver.py

```diff
@@ -0,0 +1,593 @@
+        loss :
+            The original loss function to be optimized.
+
+        Returns
+        -------
+        :
+            A function that runs the solver with the provided loss and proximal operator.
+        """
+        self._check_is_callable_from_jax(loss)
+
+        solver_kwargs = self.solver_kwargs.copy()
+        solver_kwargs["fun"] = loss
+        solver_kwargs["prox"] = self.get_prox_operator()
+
+        run_kwargs = dict(hyperparams_prox=self.regularizer_strength)
+
+        return self.get_runner(solver_kwargs, run_kwargs)
+
+
+class LassoSolver(ProxGradientSolver):
+    """
+    Solver for optimization using the Lasso (L1 regularization) method with Proximal Gradient.
+
+    This class is a specialized version of the ProxGradientSolver with the proximal operator
+    set for L1 regularization (Lasso). It utilizes the `jaxopt` library's proximal gradient optimizer.
+    """
+
+    def __init__(
+        self,
+        solver_name: str = "ProximalGradient",
+        solver_kwargs: Optional[dict] = None,
+        regularizer_strength: float = 1.0,
+        mask: Optional[Union[NDArray, jnp.ndarray]] = None,
+    ):
+        super().__init__(
+            solver_name,
+            solver_kwargs=solver_kwargs,
+            mask=mask,
+        )
+        self.regularizer_strength = regularizer_strength
+
+    def get_prox_operator(
+        self,
+    ) -> Callable[
+        [Tuple[jnp.ndarray, jnp.ndarray], float, float], Tuple[jnp.ndarray, jnp.ndarray]
+    ]:
+        """
+        Retrieve the proximal operator for Lasso regularization (L1 penalty).
+
+        Returns
+        -------
+        :
+            The proximal operator, applying L1 regularization to the provided parameters. The intercept
+            term is not regularized.
+        """
+
+        def prox_op(params, l1reg, scaling=1.0):
+            Ws, bs = params
+            return jaxopt.prox.prox_lasso(Ws, l1reg, scaling=scaling), bs
+
+        return prox_op
+
+
+class GroupLassoSolver(ProxGradientSolver):
+    """
+    Solver for optimization using the Group Lasso regularization method with Proximal Gradient.
+
+    This class is a specialized version of the ProxGradientSolver with the proximal operator
+    set for Group Lasso regularization. The Group Lasso regularization induces sparsity on groups
+    of features rather than individual features.
+
+    Attributes
+    ----------
+    mask : Union[jnp.ndarray, NDArray]
+        A mask array indicating groups of features for regularization.
+        Each row represents a group of features.
+        Each column corresponds to a feature, where a value of 1 indicates that the feature belongs
+        to the group, and a value of 0 indicates it doesn't.
+
+    Methods
+    -------
+    _check_mask():
+        Validate the mask array to ensure it meets the requirements for Group Lasso regularization.
+    get_prox_operator():
+        Retrieve the proximal operator for Group Lasso regularization.
+    """
```

Comment:
don't document methods here

Reply:
and don't bring hidden methods to user's attention at all

Reply:
removed them